unstructured/ingest/connector/sharepoint.py

```diff
@@ -35,6 +37,9 @@
 class SimpleSharepointConfig(BaseConnectorConfig):
     client_id: str
     client_credential: str = field(repr=False)
+    permissions_application_id: str
```

Comment:
Should these all be optional since it's an additional step that a user can choose to omit?

Reply:
Update: A `permissions_config` argument of type `SharepointPermissionsConfig` is made optional to `SimpleSharepointConfig` now; and these fields are kept mandatory within the `SharepointPermissionsConfig` class

Optional assignment of a `SharepointPermissionsConfig` to `SimpleSharepointConfig` is made in the runner file 

Check: https://github.com/Unstructured-IO/unstructured/pull/1628/commits/c491ff06e02938023c3896083630d5358635f6a6